server/middlewares/UserValidation.js

```diff
@@ -16,20 +17,65 @@ class UserValidation {
     * @static
     */
   static validateUserSignUp(req, res, next) {
-    let { fullName } = req.body;
-    const { avatarUrl, bio } = req.body;
-    req.checkBody('fullName', 'full name is too long').isLength({ max: 40 });
+    UserValidation.validateFullName(req);
+    UserValidation.validateEmail(req);
+    UserValidation.validatePassword(req);
+    UserValidation.validateAvatarUrl(req);
+    UserValidation.validateBio(req);
```

Comment:
This class looks much better now, good job on that.
Since we are not requiring the user to provide avatarUrl and bio at sign up, why did you add the validations here?

Reply:
We do not want to expose any vulnerability of any kind on the signup route. If for example, someone finds out that a particular field on the user table was not validated for, that might be our system's Achilles heel. The `validateAvatarUrl()` and `validateBio()` only comes to action if a someone enters some characters on those fields. 
Therefore, I find it important that this validations remain in place.

Reply:
Ok. fine

Reply:
I suggest you only validate what is required at signup. If you check the database schema at https://app.quickdatabasediagrams.com/#/d/OnZQOZ, the user table includes other fields that are not required like googleId, facebookId etc Are you saying you will still need to include them in `validateSignUp` method.

Reply:
Alternatively, what we could do is check that the object keys passed in req.body are fullName, email, password. If anything else is passed, the request is bounced.

Reply:
Thanks, teammates for the feedback. I will implement your suggestions.